File: Chef.py
T=int(input())
if T<1 or T>1000:
    exit(0)
sum = 0
for test in range (T):
    k=int(input())
    sum += k
    if k<1 or k>100000 or sum > 500000:
        exit(0)
    ni=[int(x) for x in input().split()]
    mi=[int(x) for x in input().split()]
    if min(mi) < 0 or min(ni) < 0 or max(mi) > 10**8 or max(ni) > 10**8:
        exit(0)
    i=0
    while (i+3)<=k and str(mi)!=str(ni):
        if mi[i]!=ni[i]:
            if mi[i] < ni[i]:
                break
            else:
                h=1
                for j in range(i,i+3):
                    ni[j] += h
                    h+=1
        i+=1
    if str(mi)==str(ni):
        print("TAK")
    else:
        print("NIE")

File: chef.c
#include<stdio.h>
#include<stdlib.h>
int main(){
    int T, t;
    long sum = 0, k, *ni, *mi, i, mul;
    scanf("%d",&T);
    if (T<1 || T>1000)
        exit(0);    
    for (t = 0; t<T; t++){
        scanf("%ld",&k);
        sum += k;
        if (k<1 || k>100000 || sum > 500000)
            exit(0);
        ni = (long*)malloc(k*sizeof(long));
        mi = (long*)malloc(k*sizeof(long));
        for (i = 0; i<k ; i++){
            scanf("%ld",&ni[i]);
            if (ni[i] < 0 || ni[i] > 100000000)
                exit(0);
        }
        for (i = 0; i<k ; i++){
            scanf("%ld",&mi[i]);
            if (mi[i] < 0 || mi[i] > 100000000)
                exit(0);
        }
        i=0;
        while ((i+2)<k){
            if (mi[i]!=ni[i]){
                if (mi[i] < ni[i])
                    break;
                else{
                    mul = mi[i] - ni[i];
                    ni[i] += (1*mul); ni[i+1] += (2*mul); ni[i+2] += (3*mul);
                }
            }
            i ++;
        }
        short flag = 1;
        for (i =0; i<k ;i++)
            if (mi[i]!=ni[i]){
                flag = 0;
                break;
            }
        if (flag == 1)
            printf("TAK\n");
        else
            printf("NIE\n");
    }
}